src/sentry/static/sentry/app/views/usageStats/usageStatsOrg.tsx

```diff
@@ -99,47 +95,109 @@ class UsageStatsOrganization extends AsyncComponent<Props, State> {
       dropped: string;
       filtered: string;
     };
+    error?: Error;
   } {
-    const {dataCategory} = this.state;
-
-    let sumTotal = 0;
-    let sumAccepted = 0;
-    let sumDropped = 0;
-    let sumFiltered = 0;
-
+    const cardData = {
+      total: '-',
+      accepted: '-',
+      dropped: '-',
+      filtered: '-',
+    };
     const chartData: ChartStats = {
       accepted: [],
       dropped: [],
       projected: [],
     };
 
-    // Please ignore this stub
-    for (let i = 1; i <= 31; i++) {
-      const date = `Mar ${i}`;
-
-      chartData.accepted.push({value: [date, 2000]} as any); // TODO(ts)
-      chartData.dropped.push({value: [date, 1000]} as any); // TODO(ts)
+    if (!orgStats) {
+      return {cardData, chartData};
+    }
 
-      sumTotal += 5000;
-      sumAccepted += 2000;
-      sumDropped += 1000;
-      sumFiltered += 2000;
+    try {
+      const {dataCategory} = this.props;
+      const rollup = '1d'; // TODO(org-stats)
+      const dateTimeFormat = rollup === '1d' ? 'MMM D' : 'MMM D, HH:mm';
+
+      const usageStats: UsageStat[] = orgStats.intervals.map(interval => {
+        const dateTime = moment(interval);
+
+        return {
+          date: dateTime.format(dateTimeFormat),
+          total: 0,
+          accepted: 0,
+          filtered: 0,
+          dropped: {total: 0},
+        };
+      });
+
+      orgStats.groups.forEach(group => {
+        const {outcome, category} = group.by;
+        if (category !== dataCategory) {
+          return;
+        }
+
+        const stats = this.mapSeriesToStats(dataCategory, group.series);
+        stats.forEach((stat, i) => {
+          usageStats[i][outcome] = outcome === Outcome.DROPPED ? {total: stat} : stat;
+        });
+      });
+
+      let sumTotal = 0;
+      let sumAccepted = 0;
+      let sumDropped = 0;
+      let sumFiltered = 0;
+      usageStats.forEach(stat => {
+        stat.total = stat.accepted + stat.filtered + stat.dropped.total;
+
+        // Card Data
+        sumTotal += stat.total;
+        sumAccepted += stat.accepted;
+        sumDropped += stat.dropped.total;
+        sumFiltered += stat.filtered;
```

Comment:
Have you considered adding totals to the API response and doing them serverside? We did something similar in getsentry billing usage and it helped keep the frontend simpler.

Reply:
we do return totals from the API currently, I think they would work instead of having to sum here